Add test detection and documentation

mikalsande opened the pull request:
Add runnables.scm for detecting tests and add examples to the README for how to connect tasks with tags from runnables.scm and how to add keyboard shortcut to run the test under the cursor.

The foundry tasks have been tested, the others were generated so they may or may not work. They serve as useful examples though.

When testing it if you get `Nothing to compile` when running a test via the play button in the UI or the keyboard shortcut it means the mapping is working 👍 To get things to actually work we'd have to set up Foundry in the repo and I'm not sure that is helpful since there are many testing frameworks for Solidity. 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added a Tasks section to the README with example task templates and development setup steps.

* **Examples**
  * Added a comprehensive Solidity test example exercising state, transfers, events, mint/burn, revert cases, and fuzzed transfers.

* **New Features**
  * Added editor task templates for running tests and analysis across multiple frameworks and enhanced Solidity test discovery to detect individual test functions and contract-wide tests.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mikalsande committed:
Add runnables.scm to detect Solidity tests

mikalsande committed:
Add example.test.sol as an example test

mikalsande committed:
Add info about runnables, tasks and keymaps to README

- Info about runnables.scm
- How to link tasks to runnables via tags
- How to run tasks via keyboard shortcut

zarifpour commented:
@coderabbitai review

mikalsande committed:
Add .zed/tasks.json example file

mikalsande committed:
Clean up example task

mikalsande commented on README.md:
````diff
@@ -14,6 +14,49 @@ Enhance Zed with Solidity language support, including syntax highlighting and la
+  }
+]
+```
+
````
Added `.zed/tasks.json` in c48fef7e2abfe9a1f6389d061d9199f55eab651d and cleaned up example task in README.md in 69cdeee60de443fc494fe6950ac0f770af3bf820

mikalsande committed:
Update .zed/tasks.json

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

czepluch commented:
Hey, so sorry about the slow response. Was starting a new job and things have been a bit hectic. Taking a look now.

czepluch committed:
Fix runnables tag matching and broaden test detection

- Fix tasks.json tags to match runnables.scm tags (solidity-test,
  solidity-contract-test). Previously tags like ["foundry", "test",
  "symbol"] didn't match any runnable, so no play buttons appeared.

- Restructure runnables.scm query to follow the pattern used by
  Zed's built-in language extensions.

- Broaden test function regex from ^(test_) to ^test to match all
  Solidity test conventions: test_, testFuzz_, testFail_, testFork_.

- Add separate invariant pattern for invariant test detection.

- Consolidate duplicate security/linting tasks that were identical.

czepluch commented:
Thanks for working on this! I tested the extension locally and found a couple of issues that prevented the play buttons from appearing. I pushed a fix to your branch:

**Tag mismatch**: The tags in `tasks.json` (e.g. `["foundry", "test", "symbol"]`) didn't match the tags set in `runnables.scm` (`solidity-test`, `solidity-contract-test`). Zed requires an exact match between the `#set! tag` in runnables.scm and the `tags` array in tasks.json for the play buttons to show up. Fixed all task definitions to use the correct tags.

czepluch commented:
Can you please pump the version @mikalsande?

czepluch committed:
Bump version to 0.2.2

czepluch committed:
Merge branch 'main' into runnables
